### PR TITLE
Fix unsigned integers support for pgsql

### LINKED ISSUE #3013: [PostgreSQL] Define `ulong->decimal` mapping for parameters
LinqToDB queries are not working when they need to instantiate entities with unsigned properties. Consider the following entity:
```cs
public class UserEntity
{
    [Key]
    public int Id { get; set; }
    public ulong UserId { get; set; } // <---- unique property
    public string Username { get; set; }
    public string Discriminator { get; set; }
}
```
Running a query like this:
```cs
var dbEntry = db.Users
    .Where(x => x.UserId == userId)
    .DeleteWithOutput()
    .FirstOrDefault();
```
Or a query like this
```cs
var table = db.GetTable<UserEntity>();
table.InsertOrUpdate(
    () => new UserEntity()
    { 
        Id = 1,
        UserId = contextId,
        Username = username,
        Discriminator = discriminator
    },
    y => new UserEntity()
    {
        Username = username,
        Discriminator = discriminator
    }
);
```
Causes an exception to be thrown:
```
System.InvalidCastException: Can't write CLR type System.UInt64 with handler type NumericHandler
   at lambda_method107(Closure , NpgsqlTypeHandler , Object , NpgsqlLengthCache& , NpgsqlParameter )
   at Npgsql.TypeHandling.NpgsqlSimpleTypeHandler`1.ValidateObjectAndGetLength(Object value, NpgsqlLengthCache& lengthCache, NpgsqlParameter parameter)
   at Npgsql.NpgsqlParameter.ValidateAndGetLength()
   at Npgsql.NpgsqlCommand.ValidateParameters(ConnectorTypeMapper typeMapper)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReaderAsync(CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteQueryAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber, Func`2 func, TakeSkipDelegate skipAction, TakeSkipDelegate takeAction, CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteQueryAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber, Func`2 func, TakeSkipDelegate skipAction, TakeSkipDelegate takeAction, CancellationToken cancellationToken)
   at LinqToDB.Linq.ExpressionQuery`1.GetForEachAsync(Action`1 action, CancellationToken cancellationToken)
   at LinqToDB.AsyncExtensions.ToArrayAsync[TSource](IQueryable`1 source, CancellationToken token)
   at [Business Logic]
```

In my experiments, this only happens when the property is assigned to another variable. Assigning it to a constant literal (ie. `UserId = 1234UL`) or omitting it causes the methods to work flawlessly. `BulkCopy()` and `BulkCopyAsync()` also seem to work fine regardless of what I do.

Some advice would be appreciated.

Relevant Nuget Packages:
LinqToDb.EntityFrameworkCore: v5.2.1
Npgsql: v5.0.5
Npgsql.EntityFrameworkCore.PostgreSQL: v5.0.6

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #3013

### CODE REVIEW

File: Source/LinqToDB/DataProvider/PostgreSQL/PostgreSQLMappingSchema.cs [hunk 1]
```diff
@@ -62,9 +63,24 @@ protected PostgreSQLMappingSchema(string configuration, params MappingSchema[] s
 			AddScalarType(typeof(TimeSpan?),       DataType.Interval);
 
 			// npgsql doesn't support unsigned types except byte (and sbyte)
-			SetConvertExpression<ushort?, DataParameter>(value => new DataParameter(null, value == null ? (int?    )null : value, DataType.Int32)  , false);
-			SetConvertExpression<uint?  , DataParameter>(value => new DataParameter(null, value == null ? (long?   )null : value, DataType.Int64)  , false);
-			SetConvertExpression<ulong? , DataParameter>(value => new DataParameter(null, value == null ? (decimal?)null : value, DataType.Decimal), false);
+			SetConvertExpression<ushort , DataParameter>(value => new DataParameter(null, (int)value, DataType.Int32));
```
Comment (viceroypenguin) [inline Source/LinqToDB/DataProvider/PostgreSQL/PostgreSQLMappingSchema.cs:66]: Formatting: if we're on this vertical spacing, then you should align the `DataType.IntXx` at the far right as well

File: Tests/Linq/DataProvider/PostgreSQLTests.cs [hunk 2]
```diff
@@ -55,27 +55,27 @@ public void TestParameters([IncludeDataSources(TestProvName.AllPostgreSQL)] stri
 			using (new CustomCommandProcessor(null))
 			using (var conn = new DataConnection(context))
 			{
-				Assert.That(conn.Execute<string>("SELECT :p"       , new { p = "1" }),                                Is.EqualTo("1"));
-				Assert.That(conn.Execute<int>   ("SELECT :p"       , new { p = new DataParameter { Value = 1 } }),    Is.EqualTo(1));
-				Assert.That(conn.Execute<string>("SELECT :p1"      , new { p1 = new DataParameter { Value = "1" } }), Is.EqualTo("1"));
-				Assert.That(conn.Execute<int>   ("SELECT :p1 + :p2", new { p1 = 2, p2 = 3 }),                         Is.EqualTo(5));
-				Assert.That(conn.Execute<int>   ("SELECT :p2 + :p1", new { p2 = 2, p1 = 3 }),                         Is.EqualTo(5));
-				Assert.That(conn.Execute<string>("SELECT :p"       , new { p = 1 }),                                  Is.EqualTo("1"));
+				Assert.That(conn.Execute<string>("SELECT :p", new { p = "1" }), Is.EqualTo("1"));
```
Comment (viceroypenguin) [inline Tests/Linq/DataProvider/PostgreSQLTests.cs:58]: unintended spacing change?

**Review by viceroypenguin — APPROVED** [review]
Other than whitespace, LGTM

File: Tests/Linq/DataProvider/PostgreSQLTests.cs (hunk 2 above)
Comment (MaceWindu) [reply Tests/Linq/DataProvider/PostgreSQLTests.cs:58]: image

**Review by sdanyliv — APPROVED** [review]

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**